[Automation] native-image run fails for org.apache-extras.beanshell:bsh:2.0b6

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0b6" ./gradlew clean test -Pcoordinates="org.apache-extras.beanshell:bsh:2.0b6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581007
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
ts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libawt.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libawt_headless.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libawt_xawt.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libfontmanager.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libjava.so (jdk_library_shim)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libjavajpeg.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libjvm.so (jdk_library_shim)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/liblcms.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/org.apache-extras.beanshell.bsh_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'org.apache-extras.beanshell.bsh_tests-tests' in 2m 15s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/test-results/test/testlist
Exception in thread "Image Fetcher 0" java.lang.NoSuchFieldError: java.awt.image.BufferedImage.raster
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getFieldID(JNIFunctions.java:1871)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.GetFieldID(JNIFunctions.java:455)
	at java.desktop@25.0.3/java.awt.image.BufferedImage.initIDs(Native Method)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/test-results/test/testlist] and its subfolders.
	at java.desktop@25.0.3/java.awt.image.BufferedImage.<clinit>(BufferedImage.java:285)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.3/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.3/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)
JUnit Platform on Native Image - report
----------------------------------------

Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.3/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.3/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.3/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.3/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.3/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.3/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.3/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.3/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.3/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.3/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.3/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.3/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.3/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)
sending file: /bsh/util/lib/remote.html
org_apache_extras_beanshell.bsh.BSHAllocationExpressionTest > allocatesPrimitiveArrayWithDefinedDimensions() SUCCESSFUL

org_apache_extras_beanshell.bsh.BSHAllocationExpressionTest > allocatesObjectArrayWithUndefinedTrailingDimension() SUCCESSFUL

org_apache_extras_beanshell.bsh.BSHArrayInitializerTest > evaluatesTypedPrimitiveArrayInitializer() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassLoaderTest > delegatesClassLookupToDesignatedReloadLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > resolvesClassesWithDefaultClassForNameLookup() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > readsResourcesFromConfiguredClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > readsResourceStreamsFromConfiguredClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > resolvesClassesWithConfiguredClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > readsResourcesFromDefaultClassResourceLookup() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassBrowserTest > displaysDeclaredPublicConstructorsMethodsAndFieldsForSelectedClass() FAILED

org_apache_extras_beanshell.bsh.ClassGeneratorUtilTest > generatesClassWithScriptedConstructorAndMethod() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassGeneratorUtilTest > startsInterpreterFromClassRelativeScriptResource() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > resolvesBeanshellCoreClassesWithInterpreterClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > resolvesClassesThroughExternalClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > resolvesClassesThroughConfiguredBaseLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > consultsBaseLoaderWhenResolvingResources() SUCCESSFUL

org_apache_extras_beanshell.bsh.CollectionIteratorTest > iteratesOverJavaIterableUsingPublicIteratorContract() SUCCESSFUL

org_apache_extras_beanshell.bsh.HttpdConnectionTest > servesBundledRemoteConsoleResource() SUCCESSFUL

org_apache_extras_beanshell.bsh.InterpreterTest > invokesResolvedStaticMainMethod(Path) SUCCESSFUL

org_apache_extras_beanshell.bsh.LHSTest > readsFieldValueBeforeCompoundAssignment() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > readsObjectPropertiesThroughGetterFallbacks() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > readsPublicAndPrivateFieldsThroughBeanShellAccess() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > invokesPublicAndPrivateMethodsThroughBeanShellAccess() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > constructsObjectsWithPublicAndAccessibleConstructorLookup() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > writesObjectPropertyThroughSetterFallback() SUCCESSFUL

org_apache_extras_beanshell.bsh.XThisTest > createsAndCachesInterfaceProxyForScriptedObject() SUCCESSFUL


Failures (1):
  JUnit Jupiter:ClassBrowserTest:displaysDeclaredPublicConstructorsMethodsAndFieldsForSelectedClass()
    MethodSource [className = 'org_apache_extras_beanshell.bsh.ClassBrowserTest', methodName = 'displaysDeclaredPublicConstructorsMethodsAndFieldsForSelectedClass', methodParameterTypes = '']
    => java.lang.Error: java.home property not set
       java.desktop@25.0.3/sun.awt.FontConfiguration.findFontConfigFile(FontConfiguration.java:166)
       java.desktop@25.0.3/sun.awt.FontConfiguration.<init>(FontConfiguration.java:95)
       java.desktop@25.0.3/sun.font.MFontConfiguration.<init>(MFontConfiguration.java:48)
       java.desktop@25.0.3/sun.awt.X11FontManager.createFontConfiguration(X11FontManager.java:689)
       java.desktop@25.0.3/sun.font.SunFontManager.<init>(SunFontManager.java:341)
       java.desktop@25.0.3/sun.awt.FcFontManager.<init>(FcFontManager.java:35)
       java.desktop@25.0.3/sun.awt.X11FontManager.<init>(X11FontManager.java:55)
       java.desktop@25.0.3/sun.font.PlatformFontInfo.createFontManager(PlatformFontInfo.java:37)
       java.desktop@25.0.3/sun.font.FontManagerFactory.getInstance(FontManagerFactory.java:51)
       java.desktop@25.0.3/sun.font.SunFontManager.getInstance(SunFontManager.java:240)
       [...]

Test run finished after 24 ms
[        15 containers found      ]
[         0 containers skipped    ]
[        15 containers started    ]
[         0 containers aborted    ]
[        15 containers successful ]
[         0 containers failed     ]
[        26 tests found           ]
[         0 tests skipped         ]
[        26 tests started         ]
[         0 tests aborted         ]
[        25 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/org.apache-extras.beanshell.bsh_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 19s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache-extras.beanshell:bsh:2.0b6 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 21s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0b6 at 2026-05-12T10:39:58Z with exit code 1 after 142s
❌ ERROR: 'native-image run' for 2.0b6 failed with exit code 1.
FAILED[native-image run][2.0b6][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0b6" ./gradlew clean test -Pcoordinates="org.apache-extras.beanshell:bsh:2.0b6"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0b6" ./gradlew clean test -Pcoordinates="org.apache-extras.beanshell:bsh:2.0b6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581071
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
e/nativeTestCompile/libawt.so (jdk_library, 902.46KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libawt_headless.so (jdk_library, 41.65KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libawt_xawt.so (jdk_library, 508.08KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libfontmanager.so (jdk_library, 1.90MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libjava.so (jdk_library_shim, 9.02KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libjavajpeg.so (jdk_library, 234.34KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libjvm.so (jdk_library_shim, 9.02KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/liblcms.so (jdk_library, 524.13KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/org.apache-extras.beanshell.bsh_tests-tests (executable, 50.00MiB)
 /tmp/native-image-build-output.json (build_info, 1.05KiB)
========================================================================================================================
Finished generating 'org.apache-extras.beanshell.bsh_tests-tests' in 2m 9s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

Exception in thread "Image Fetcher 0" java.lang.NoSuchFieldError: java.awt.image.BufferedImage.raster
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getFieldID(JNIFunctions.java:1876)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.GetFieldID(JNIFunctions.java:456)
	at java.desktop@25.0.2/java.awt.image.BufferedImage.initIDs(Native Method)
	at java.desktop@25.0.2/java.awt.image.BufferedImage.<clinit>(BufferedImage.java:285)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
sending file: /bsh/util/lib/remote.html
org_apache_extras_beanshell.bsh.BSHAllocationExpressionTest > allocatesPrimitiveArrayWithDefinedDimensions() SUCCESSFUL

org_apache_extras_beanshell.bsh.BSHAllocationExpressionTest > allocatesObjectArrayWithUndefinedTrailingDimension() SUCCESSFUL

org_apache_extras_beanshell.bsh.BSHArrayInitializerTest > evaluatesTypedPrimitiveArrayInitializer() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassLoaderTest > delegatesClassLookupToDesignatedReloadLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > resolvesClassesWithDefaultClassForNameLookup() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > readsResourcesFromConfiguredClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > readsResourceStreamsFromConfiguredClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > resolvesClassesWithConfiguredClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > readsResourcesFromDefaultClassResourceLookup() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassBrowserTest > displaysDeclaredPublicConstructorsMethodsAndFieldsForSelectedClass() FAILED

org_apache_extras_beanshell.bsh.ClassGeneratorUtilTest > generatesClassWithScriptedConstructorAndMethod() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassGeneratorUtilTest > startsInterpreterFromClassRelativeScriptResource() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > resolvesBeanshellCoreClassesWithInterpreterClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > resolvesClassesThroughExternalClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > resolvesClassesThroughConfiguredBaseLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > consultsBaseLoaderWhenResolvingResources() SUCCESSFUL

org_apache_extras_beanshell.bsh.CollectionIteratorTest > iteratesOverJavaIterableUsingPublicIteratorContract() SUCCESSFUL

org_apache_extras_beanshell.bsh.HttpdConnectionTest > servesBundledRemoteConsoleResource() SUCCESSFUL

org_apache_extras_beanshell.bsh.InterpreterTest > invokesResolvedStaticMainMethod(Path) SUCCESSFUL

org_apache_extras_beanshell.bsh.LHSTest > readsFieldValueBeforeCompoundAssignment() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > readsObjectPropertiesThroughGetterFallbacks() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > readsPublicAndPrivateFieldsThroughBeanShellAccess() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > invokesPublicAndPrivateMethodsThroughBeanShellAccess() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > constructsObjectsWithPublicAndAccessibleConstructorLookup() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > writesObjectPropertyThroughSetterFallback() SUCCESSFUL

org_apache_extras_beanshell.bsh.XThisTest > createsAndCachesInterfaceProxyForScriptedObject() SUCCESSFUL


Failures (1):
  JUnit Jupiter:ClassBrowserTest:displaysDeclaredPublicConstructorsMethodsAndFieldsForSelectedClass()
    MethodSource [className = 'org_apache_extras_beanshell.bsh.ClassBrowserTest', methodName = 'displaysDeclaredPublicConstructorsMethodsAndFieldsForSelectedClass', methodParameterTypes = '']
    => java.lang.Error: java.home property not set
       java.desktop@25.0.2/sun.awt.FontConfiguration.findFontConfigFile(FontConfiguration.java:166)
       java.desktop@25.0.2/sun.awt.FontConfiguration.<init>(FontConfiguration.java:95)
       java.desktop@25.0.2/sun.font.MFontConfiguration.<init>(MFontConfiguration.java:48)
       java.desktop@25.0.2/sun.awt.X11FontManager.createFontConfiguration(X11FontManager.java:689)
       java.desktop@25.0.2/sun.font.SunFontManager.<init>(SunFontManager.java:341)
       java.desktop@25.0.2/sun.awt.FcFontManager.<init>(FcFontManager.java:35)
       java.desktop@25.0.2/sun.awt.X11FontManager.<init>(X11FontManager.java:55)
       java.desktop@25.0.2/sun.font.PlatformFontInfo.createFontManager(PlatformFontInfo.java:37)
       java.desktop@25.0.2/sun.font.FontManagerFactory.getInstance(FontManagerFactory.java:51)
       java.desktop@25.0.2/sun.font.SunFontManager.getInstance(SunFontManager.java:240)
       [...]

Test run finished after 24 ms
[        15 containers found      ]
[         0 containers skipped    ]
[        15 containers started    ]
[         0 containers aborted    ]
[        15 containers successful ]
[         0 containers failed     ]
[        26 tests found           ]
[         0 tests skipped         ]
[        26 tests started         ]
[         0 tests aborted         ]
[        25 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/org.apache-extras.beanshell.bsh_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 13s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache-extras.beanshell:bsh:2.0b6 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 15s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0b6 at 2026-05-12T10:39:00Z with exit code 1 after 135s
❌ ERROR: 'native-image run' for 2.0b6 failed with exit code 1.
FAILED[native-image run][2.0b6][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0b6" ./gradlew clean test -Pcoordinates="org.apache-extras.beanshell:bsh:2.0b6"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.0b6" ./gradlew clean test -Pcoordinates="org.apache-extras.beanshell:bsh:2.0b6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581030
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
so (jdk_library, 902.46KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libawt_headless.so (jdk_library, 41.65KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libawt_xawt.so (jdk_library, 508.08KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libfontmanager.so (jdk_library, 1.90MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libjava.so (jdk_library_shim, 9.02KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libjavajpeg.so (jdk_library, 234.34KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libjvm.so (jdk_library_shim, 9.02KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/liblcms.so (jdk_library, 524.13KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/org.apache-extras.beanshell.bsh_tests-tests (executable, 50.25MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'org.apache-extras.beanshell.bsh_tests-tests' in 2m 14s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

sending file: /bsh/util/lib/remote.html
org_apache_extras_beanshell.bsh.BSHAllocationExpressionTest > allocatesPrimitiveArrayWithDefinedDimensions() SUCCESSFUL

org_apache_extras_beanshell.bsh.BSHAllocationExpressionTest > allocatesObjectArrayWithUndefinedTrailingDimension() SUCCESSFUL

org_apache_extras_beanshell.bsh.BSHArrayInitializerTest > evaluatesTypedPrimitiveArrayInitializer() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassLoaderTest > delegatesClassLookupToDesignatedReloadLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > resolvesClassesWithDefaultClassForNameLookup() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > readsResourcesFromConfiguredClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > readsResourceStreamsFromConfiguredClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > resolvesClassesWithConfiguredClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.BshClassManagerTest > readsResourcesFromDefaultClassResourceLookup() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassBrowserTest > displaysDeclaredPublicConstructorsMethodsAndFieldsForSelectedClass() FAILED

org_apache_extras_beanshell.bsh.ClassGeneratorUtilTest > generatesClassWithScriptedConstructorAndMethod() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassGeneratorUtilTest > startsInterpreterFromClassRelativeScriptResource() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > resolvesBeanshellCoreClassesWithInterpreterClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > resolvesClassesThroughExternalClassLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > resolvesClassesThroughConfiguredBaseLoader() SUCCESSFUL

org_apache_extras_beanshell.bsh.ClassManagerImplTest > consultsBaseLoaderWhenResolvingResources() SUCCESSFUL

org_apache_extras_beanshell.bsh.CollectionIteratorTest > iteratesOverJavaIterableUsingPublicIteratorContract() SUCCESSFUL

org_apache_extras_beanshell.bsh.HttpdConnectionTest > servesBundledRemoteConsoleResource() SUCCESSFUL

org_apache_extras_beanshell.bsh.InterpreterTest > invokesResolvedStaticMainMethod(Path) SUCCESSFUL

org_apache_extras_beanshell.bsh.LHSTest > readsFieldValueBeforeCompoundAssignment() SUCCESSFUL
Exception in thread "Image Fetcher 0" java.lang.NoSuchFieldError: java.awt.image.BufferedImage.raster
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getFieldID(JNIFunctions.java:1876)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.GetFieldID(JNIFunctions.java:456)
	at java.desktop@25.0.2/java.awt.image.BufferedImage.initIDs(Native Method)
	at java.desktop@25.0.2/java.awt.image.BufferedImage.<clinit>(BufferedImage.java:285)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Exception in thread "Image Fetcher 1" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Exception in thread "Image Fetcher 0" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Exception in thread "Image Fetcher 1" java.lang.NoClassDefFoundError: Could not initialize class java.awt.image.BufferedImage
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createImage(ImageRepresentation.java:180)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.createBufferedImage(ImageRepresentation.java:238)
	at java.desktop@25.0.2/sun.awt.image.ImageRepresentation.setPixels(ImageRepresentation.java:339)
	at java.desktop@25.0.2/sun.awt.image.ImageDecoder.setPixels(ImageDecoder.java:126)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.sendPixels(GifImageDecoder.java:447)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.parseImage(Native Method)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.readImage(GifImageDecoder.java:596)
	at java.desktop@25.0.2/sun.awt.image.GifImageDecoder.produceImage(GifImageDecoder.java:212)
	at java.desktop@25.0.2/sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:247)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:212)
	at java.desktop@25.0.2/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)

org_apache_extras_beanshell.bsh.ReflectTest > readsObjectPropertiesThroughGetterFallbacks() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > readsPublicAndPrivateFieldsThroughBeanShellAccess() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > invokesPublicAndPrivateMethodsThroughBeanShellAccess() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > constructsObjectsWithPublicAndAccessibleConstructorLookup() SUCCESSFUL

org_apache_extras_beanshell.bsh.ReflectTest > writesObjectPropertyThroughSetterFallback() SUCCESSFUL

org_apache_extras_beanshell.bsh.XThisTest > createsAndCachesInterfaceProxyForScriptedObject() SUCCESSFUL


Failures (1):
  JUnit Jupiter:ClassBrowserTest:displaysDeclaredPublicConstructorsMethodsAndFieldsForSelectedClass()
    MethodSource [className = 'org_apache_extras_beanshell.bsh.ClassBrowserTest', methodName = 'displaysDeclaredPublicConstructorsMethodsAndFieldsForSelectedClass', methodParameterTypes = '']
    => java.lang.Error: java.home property not set
       java.desktop@25.0.2/sun.awt.FontConfiguration.findFontConfigFile(FontConfiguration.java:166)
       java.desktop@25.0.2/sun.awt.FontConfiguration.<init>(FontConfiguration.java:95)
       java.desktop@25.0.2/sun.font.MFontConfiguration.<init>(MFontConfiguration.java:48)
       java.desktop@25.0.2/sun.awt.X11FontManager.createFontConfiguration(X11FontManager.java:689)
       java.desktop@25.0.2/sun.font.SunFontManager.<init>(SunFontManager.java:341)
       java.desktop@25.0.2/sun.awt.FcFontManager.<init>(FcFontManager.java:35)
       java.desktop@25.0.2/sun.awt.X11FontManager.<init>(X11FontManager.java:55)
       java.desktop@25.0.2/sun.font.PlatformFontInfo.createFontManager(PlatformFontInfo.java:37)
       java.desktop@25.0.2/sun.font.FontManagerFactory.getInstance(FontManagerFactory.java:51)
       java.desktop@25.0.2/sun.font.SunFontManager.getInstance(SunFontManager.java:240)
       [...]

Test run finished after 26 ms
[        15 containers found      ]
[         0 containers skipped    ]
[        15 containers started    ]
[         0 containers aborted    ]
[        15 containers successful ]
[         0 containers failed     ]
[        26 tests found           ]
[         0 tests skipped         ]
[        26 tests started         ]
[         0 tests aborted         ]
[        25 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build/native/nativeTestCompile/org.apache-extras.beanshell.bsh_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 18s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache-extras.beanshell:bsh:2.0b6 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 20s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0b6 at 2026-05-12T10:40:12Z with exit code 1 after 141s
❌ ERROR: 'native-image run' for 2.0b6 failed with exit code 1.
FAILED[native-image run][2.0b6][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.0b6" ./gradlew clean test -Pcoordinates="org.apache-extras.beanshell:bsh:2.0b6"]
```

--- a/metadata/org.apache-extras.beanshell/bsh/2.0b6/reachability-metadata.json
+++ b/metadata/org.apache-extras.beanshell/bsh/2.0b6/reachability-metadata.json
@@ -1959,6 +1959,50 @@
       "condition" : {
         "typeReached" : "bsh.util.ClassBrowser"
       },
+      "type" : "java.awt.image.BufferedImage",
+      "jniAccessible" : true,
+      "fields" : [
+        {
+          "name" : "colorModel"
+        },
+        {
+          "name" : "imageType"
+        },
+        {
+          "name" : "raster"
+        }
+      ],
+      "methods" : [
+        {
+          "name" : "getRGB",
+          "parameterTypes" : [
+            "int",
+            "int",
+            "int",
+            "int",
+            "int[]",
+            "int",
+            "int"
+          ]
+        },
+        {
+          "name" : "setRGB",
+          "parameterTypes" : [
+            "int",
+            "int",
+            "int",
+            "int",
+            "int[]",
+            "int",
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "bsh.util.ClassBrowser"
+      },
       "type" : "sun.java2d.Disposer",
       "jniAccessible" : true,
       "methods" : [

--- a/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build.gradle
+++ b/tests/src/org.apache-extras.beanshell/bsh/2.0b6/build.gradle
@@ -15,6 +15,11 @@ dependencies {
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 
+tasks.named("nativeTest") {
+    runtimeArgs.add("-Djava.home=${System.getenv('JAVA_HOME')}")
+    runtimeArgs.add("-Djava.class.path=${sourceSets.test.runtimeClasspath.asPath}")
+}
+
 graalvmNative {
     binaries {
         test {

